src/common/transformations/src/transformations/common_optimizations/transpose_sinking_unary.cpp

```diff
@@ -0,0 +1,141 @@
+    first_node_inputs[0] = new_first_node;
+    auto new_second_node = first_node->clone_with_new_inputs(first_node_inputs);
+
+    new_second_node->set_friendly_name(second_node->get_friendly_name());
+    ov::copy_runtime_info({first_node, second_node}, {new_first_node, new_second_node});
+
+    ov::replace_node(second_node, new_second_node);
+
+    return std::make_pair(new_first_node, new_second_node);
+}
+
+/**
+ * @brief SwapOutputs has much better performance than SwapNodes and covers the most of the real situations
+ *        but cannot work when the consumers count greater than one
+ * @param first_node first node pointer
+ * @param second_node second node pointer
+ * @return NodePair pair of nodes in new order that allows to register them in MatcherPass
+ */
+NodePair SwapOutputs(NodePtr first_node, NodePtr second_node) {
+    const auto first_node_output_names = first_node->output(0).get_names();
+    const auto second_node_output_names = second_node->output(0).get_names();
+
+    auto swap_names = [&]() {
+        const std::string first_name = first_node->get_friendly_name();
+        first_node->set_friendly_name(second_node->get_friendly_name());
+        second_node->set_friendly_name(first_name);
+
+        first_node->output(0).set_names(second_node_output_names);
+        second_node->output(0).set_names(first_node_output_names);
+    };
+
+    auto out_1 = first_node->input_value(0);
+    second_node->input(0).replace_source_output(out_1);
+
+    auto out_2 = second_node->output(0);
+    second_node->output(0).replace(first_node->output(0));
+
+    first_node->input(0).replace_source_output(out_2);
+
+    swap_names();
+
+    return std::make_pair(second_node, first_node);
+}
+
+}  // namespace
+
+ov::pass::TransposeSinkingUnaryForward::TransposeSinkingUnaryForward() {
+    MATCHER_SCOPE(TransposeSinkingUnaryForward);
+
+    auto transpose_label = ov::pass::pattern::wrap_type<ov::opset9::Transpose>(
+        {ov::pass::pattern::any_input(), ov::pass::pattern::any_input()});
+    auto unary_label = ov::pass::pattern::wrap_type<ov::op::util::UnaryElementwiseArithmetic,
+                                                    ov::opset9::Clamp,
+                                                    ov::opset9::Elu,
+                                                    ov::opset9::SoftPlus,
+                                                    ov::opset9::LogicalNot,
+                                                    ov::opset9::Convert>({transpose_label});
+
+    ov::matcher_pass_callback matcher_pass_callback = [=](ov::pass::pattern::Matcher& m) {
+        const auto& pattern_to_output = m.get_pattern_value_map();
+        auto transpose = pattern_to_output.at(transpose_label).get_node_shared_ptr();
+        auto unary = pattern_to_output.at(unary_label).get_node_shared_ptr();
+
+        NodePair new_nodes;
+
+        if (transpose->output(0).get_target_inputs().size() > 1 || unary->output(0).get_target_inputs().size() > 1)
```

Comment:
There is a sense to have this condition in one place for both passes, in `Swap` function, for example